core/block_test.go

```diff
@@ -177,3 +179,190 @@ func TestBlockHash(t *testing.T) {
+		switch t {
+		case "DECLARE":
+			return Declare
+		case "DEPLOY":
+			return Deploy
+		case "DEPLOY_ACCOUNT":
+			return DeployAccount
+		case "INVOKE_FUNCTION":
+			return Invoke
+		case "L1_HANDLER":
+			return L1Handler
+		default:
+			return -1
+		}
+	}
+
+	for i, r := range receiptsInterface {
+		receipt := r.(map[string]interface{})
+		txn := txns[i].(map[string]interface{})
+		var events []*Event
+		for _, e := range receipt["events"].([]interface{}) {
+			event := e.(map[string]interface{})
+			var data []*felt.Felt
+			for _, d := range event["data"].([]interface{}) {
+				data = append(data, hexToFelt(d.(string)))
+			}
+			var keys []*felt.Felt
+			for _, k := range event["keys"].([]interface{}) {
+				keys = append(keys, hexToFelt(k.(string)))
+			}
+			events = append(events, &Event{
+				Data: data,
+				From: hexToFelt(event["from_address"].(string)),
+				Keys: keys,
+			})
+		}
+		var signatures []*felt.Felt
+		if txn["signature"] != nil {
+			for _, s := range txn["signature"].([]interface{}) {
+				signatures = append(signatures, hexToFelt(s.(string)))
+			}
+		}
+		// Some of these values are set to nil since they are not required to calculate the commitment.
+		transactionReceipt := TransactionReceipt{
+			Events:          events,
+			Signatures:      signatures,
+			TransactionHash: hexToFelt(receipt["transaction_hash"].(string)),
+			Type:            transactionType(txn["type"].(string)),
+		}
+		receipts[i] = &transactionReceipt
+	}
+	return receipts
+}
+
+func init() {
+	var t *testing.T
+	getTransactionReceipts(t)
+}
+
+func TestTransactionCommitment(t *testing.T) {
+	tests := []struct {
+		txs  []*TransactionReceipt
+		want string
+	}{
+		{
+			receipts[0],
+			"0x24638e0ca122d0260d54e901dc0942ea68bd1fc40a96b5da765985c47c92500",
+		},
+		{
+			receipts[1],
+			"0x18bb7d6c1c558aa0a025f08a7d723a44b13008ffb444c432077f319a7f4897c",
+		},
+		{
+			receipts[2],
+			"0xbf11745df434cbd284e13ca36354139a4bca2f6722e737c6136590990c8619",
+		},
+		{
+			receipts[3],
+			"0x580a06bfc8c3fe39bbb7c5d16298b8928bf7c28f4c31b8e6b48fc25cd644fc1",
+		},
+	}
+
+	for _, test := range tests {
+		commitment, _ := TransactionCommitment(test.txs)
+		if "0x"+commitment.Text(16) != test.want {
+			t.Errorf("got %s, want %s", "0x"+commitment.Text(16), test.want)
+		}
+	}
+}
+
+func TestEventCommitment(t *testing.T) {
+	tests := []struct {
+		receipts []*TransactionReceipt
+		want     string
+	}{
+		{
+			receipts[0],
+			"0x5d25e41d43b00681cc63ed4e13a82efe3e02f47e03173efbd737dd52ba88c7e",
+		},
+		{
+			receipts[1],
+			"0x0",
+		},
+		{
+			receipts[2],
+			"0x0",
+		},
+		{
+			receipts[3],
+			"0x6f499789aabb31935810ce89d6ea9e9d37c5921c0d7fae2bd68f2fff5b7b93f",
+		},
+	}
+
+	for _, test := range tests {
+		eventData, _ := ComputeEventData(test.receipts)
+		if "0x"+(eventData.EventCommitment).Text(16) != test.want {
+			t.Errorf("got %s, want %s", "0x"+(eventData.EventCommitment).Text(16), test.want)
+		}
+	}
```

Comment:
We can have a subtest for each case, giving us more informative messages like "receipt1Goerli".